modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Filters.tsx

```diff
@@ -75,73 +77,104 @@ function alertSuccess() {
 	});
 }
 
+interface IPropsAddFDSFilterModalContent {
+	closeModal: Function;
+	fdsView: FDSViewType;
+	fields: IField[];
+	filter?: IDateFilter | IDynamicFilter;
+	namespace: string;
+	onSave: (newFilter: IFilter) => void;
+}
+
 function AddFDSFilterModalContent({
 	closeModal,
 	fdsView,
 	fields,
+	filter,
 	namespace,
 	onSave,
 }: IPropsAddFDSFilterModalContent) {
-	const fromFormElementId = `${namespace}From`;
-	const formElementId = `${namespace}Form`;
-	const toFormElementId = `${namespace}To`;
-	const nameFormElementId = `${namespace}Name`;
-	const selectedFieldFormElementId = `${namespace}SelectedField`;
-	const sourceOptionFormElementId = `${namespace}SourceOption`;
-	const multipleFormElementId = `${namespace}Multiple`;
-	const includeModeFormElementId = `${namespace}IncludeMode`;
-	const preselectedValuesFormElementId = `${namespace}PreselectedValues`;
-
-	const [formValues, setFormValues] = useState<{[key: string]: string}>({
-		[fromFormElementId]: '',
-		[includeModeFormElementId]: 'true',
-		[multipleFormElementId]: 'true',
-		[nameFormElementId]: '',
-		[preselectedValuesFormElementId]: '',
-		[selectedFieldFormElementId]: '',
-		[sourceOptionFormElementId]: '',
-		[toFormElementId]: '',
-	});
-	const [multiSelectValues, setMultiSelectValues] = useState<any[]>([]);
+	const [from, setFrom] = useState<string>(
+		(filter as IDateFilter)?.from ?? ''
+	);
+	const [include, setInclude] = useState<boolean>(
+		(filter as IDynamicFilter)?.include ?? true
+	);
+	const [isValidDateRange, setIsValidDateRange] = useState(true);
```

Comment:
[nit] I'd call this `dateRangeValid` and `included`, to keep the most important last word adjective, that fits booleans. It's a bit odd to see the `is` prefix, but I guess it's fine in React, as there is no  `isIs*` getter.